### DISCUSSION TITLE
Speeding up large runs

### DISCUSSION THREAD

**Discussion by TravisHaussler** [synthetichealth/synthea#1475] [General]
Is there any documentation or idea list for how to run large population sets (1000s of patients, perhaps even 100k) by trimming options or suppressing certain operations that might not be needed?

**Comment by jawalonoski** [comment]
The documentation you're looking for is not consolidated in one place. It has also changed over time, and may continue to do so.

- You can change the memory available to your JVM.

- You can change the number of threads in use, but the default is already "good."

For example, from `synthea.properties`:
```properties
# the number of threads to use for the generator, set the value to -1 to match the number of
# available processors (as per Runtime.getRuntime().availableProcessors())
# defaults to -1 if not specified
generate.thread_pool_size = -1
```

- From the default settings, you can turn off console output which could save time based on your platform.
```properties
generate.log_patients.detail = simple
# options are "none", "simple", or "detailed" (without quotes). defaults to simple if another value is used
# none = print nothing to the console during generation
```
- You can also turn off exporters, which eliminates I/O. Of course, you'd end up with no output data, but you could theoretically design an architecture that passed finished patients off to another process to handle export. There is no built-in support for that.

- For XML or JSON export, you could turn off pretty printing... which might save time over a significant number of patients.

```properties
# exporters that use XML or JSON can enable or disable 'pretty printing'
exporter.pretty_print = true
```

There are probably other ideas, but that is what I have off the top of my head.

**Comment by TravisHaussler** [comment]
Thanks, I will give these a try. Good idea on the VM settings, I will try increasing memory allocation.   I am also now trying to get a [keep module](https://github.com/synthetichealth/synthea/wiki/Keep-Patients-Module) working to see if that makes it more efficient to generate patients I might care about in one study or another
